sdk/tables/Azure.Data.Tables/tests/TableConnectionStringTests.cs

```diff
@@ -115,21 +115,48 @@ public void ParsesCosmos(string connString)
             Assert.That(tcs.TableStorageUri.SecondaryUri, Is.EqualTo(new Uri($"https://{AccountName}{TableConstants.ConnectionStrings.SecondaryLocationAccountSuffix}.table.cosmos.azure.com:443/")), "The SecondaryUri should have matched.");
         }
 
+        public static IEnumerable<object[]> ValidSaSStorageConnStrings()
+        {
+            yield return new object[] { $"DefaultEndpointsProtocol=https;AccountName={AccountName};SharedAccessSignature={SasToken};EndpointSuffix=core.windows.net" };
+            yield return new object[] { $"AccountName={AccountName};SharedAccessSignature={SasToken};EndpointSuffix=core.windows.net" };
+            yield return new object[] { $"DefaultEndpointsProtocol=https;SharedAccessSignature={SasToken};AccountName={AccountName}" };
+            yield return new object[] { $"AccountName={AccountName};SharedAccessSignature={SasToken}" };
+            yield return new object[] { $"BlobEndpoint=https://{AccountName}.blob.core.windows.net/;QueueEndpoint=https://{AccountName}.queue.core.windows.net/;FileEndpoint=https://{AccountName}.file.core.windows.net/;TableEndpoint=https://{AccountName}.table.core.windows.net/;SharedAccessSignature={SasToken}" };
+        }
+
         /// <summary>
         /// Validates the functionality of the TableConnectionString.
         /// </summary>
         [Test]
-        public void ParsesSaS()
+        [TestCaseSource(nameof(ValidSaSStorageConnStrings))]
+        public void ParsesSaSStorage(string connString)
```

Comment:
```suggestion
        public void ParsesSasStorage(string connString)
```

?   Not a bit deal, just looked weird to me.  😄